### PR TITLE
[FEATURE] Move exposed access token to environment variables

### PR DESCRIPTION
**Pull request by duydl**
Current code to push to hub is not best practice.
```py
# Set to true if you want to save to the huggingface hub
if True:
    trainer.push_to_hub(token="hf_123abc321bca", tags=finetune_tags)
```

 I made some small changes to load secret from  .env. 
 Perhaps it would be convenient to add to other modules also.

### CODE REVIEW

File: 1_instruction_tuning/notebooks/sft_finetuning_example.ipynb [hunk 1]
```diff
@@ -59,7 +68,7 @@
     "model, tokenizer = setup_chat_format(model=model, tokenizer=tokenizer)\n",
     "\n",
     "# Set our name for the finetune to be saved &/ uploaded to\n",
-    "finetune_name = \"SmolLM2-FT-MyDataset\"\n",
```
Comment (burtenshaw) [inline 1_instruction_tuning/notebooks/sft_finetuning_example.ipynb:62]: I think this was ok with a sufiix because it encourages people to change.

**Comment by duydl** [comment]
@burtenshaw Do you think this approach approriate? I decide not to use login() as the kernel keep running after the login prompt appears and could reach an error before I could enter the token. I think just using `input()` to set the environment var is sufficient.

**Comment by duydl** [comment]
@burtenshaw I think this would work well enough. Just setting the env var HF_TOKEN is sufficient for authen. The variable would persist through kernel restart on Colab and Kaggle in .env file.

**Comment by burtenshaw** [comment]
@duydl thanks for going through this. I'm going to merge it in it's current state. I know it's not as you implemented bet we should not write to text files in notebook envs where the interpreter is exposed. It basically means that if someone shared the notebook, they would share their hf token. 

The huggingface_hub library deals with that for use. You can use a traditional local environment variable, and `login()` will find it.

**Comment by duydl** [comment]
@burtenshaw No problem. Though I didn't know that the text file in a session would persist in Colab though I never used it. I believe that it doesn't on Kaggle. Kind of unexpected.

**Pull request merged**